build.gradle

```diff
@@ -235,76 +236,99 @@ java {
-    }
-    gameTestClient {
-        configure('client')
-        modSources.add((SourceSet) sourceSets.gameTest)
-    }
-    junit {
-        unitTestSources.add((SourceSet) sourceSets.test)
+    for (String name : secondaryModules) {
+        mods.maybeCreate(name).modSourceSets.add(sourceSets.named(name))
     }
-    data {
-        programArguments.addAll((String[]) ['--all', '--output', file('src/datagen/generated/').absolutePath,
-                                            '--mod', 'mekanism', '--existing', file('src/main/resources/').absolutePath])
 
-        modSources.add((SourceSet) sourceSets.datagenMain)
+    accessTransformers = [
+            'src/main/resources/META-INF/accesstransformer.cfg',
+            'src/additions/resources/META-INF/accesstransformer.cfg',
+            //Dev time only ATs so the file name doesn't have to match accesstransformer.cfg
+            'src/datagen/main/resources/META-INF/datagen_ats.cfg',
+            'src/gameTest/resources/META-INF/gametest_ats.cfg',
+    ]
 
-        for (String name : secondaryModules) {
-            modSources.add(getExtraSourceSet(name, 'datagen').get())
-            programArguments.addAll((String[]) ['--mod', "mekanism${name}", '--existing', file("src/${name}/resources/").absolutePath])
-        }
+    runs {
+        configureEach {
+            if (hasProperty('forge_force_ansi')) {
+                //Force ansi if declared as a gradle variable, as the auto detection doesn't detect IntelliJ properly
+                // or eclipse's plugin that adds support for ansi escape in console
+                systemProperties.put('terminal.ansi', (String) property('forge_force_ansi'))
+            }
+
+            loadedMods = [mods.mekanism]
+            for (String name : secondaryModules) {
+                loadedMods.add(mods.named(name))
+            }
 
-        dependencies {
-            runtime(configurations.datagenNonMod)
+            //if the selected toolchain is a JBR, enable DCEVM
+            if (javaToolchains.launcherFor(java.toolchain).map { it.metadata.vendor }.getOrElse('').contains('JetBrains')) {
+                jvmArguments.addAll('-XX:+IgnoreUnrecognizedVMOptions', '-XX:+AllowEnhancedClassRedefinition')
+            }
+        }
+        //Note: To enable logging into the client account, set the enable_devlogin property to true in your gradle user home
+        client {
+            client()
+            devLogin = Boolean.valueOf(project.enable_devlogin)
+        }
+        clientAlt {
+            client()
+            // Force dev login to be disabled for this run to allow using a secondary username
+            devLogin = false
+            programArguments.addAll('--username', 'AltDev')
+        }
+        server {
+            server()
+        }
+        gameTestServer {
+            type = "gameTestServer"
+            sourceSet = sourceSets.gameTest
+            loadedMods = neoForge.mods
+        }
+        gameTestClient {
```

Comment:
You can add the argument you mentioned to me to this:
`programArguments.addAll("--quickPlaySingleplayer", "GameTestWorld")`